### PR TITLE
Upgraded to OpenSearch 1.2.3.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

- Upgraded to OpenSearch 1.2.3
- Expanded build matrix to MacOS
- Added bin to .gitignore
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by dblock** [comment]
I don't get how I'm supposed to fix this or why this happens. Adding `import de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin` doesn't seem to do anything?

```
* What went wrong:
A problem occurred evaluating root project 'plugin-template'.
> Failed to apply plugin class 'de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin'.
   > Could not create plugin of type 'ForbiddenApisPlugin'.
      > Could not initialize class de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin
```

**Comment by saratvemulapalli** [comment]
> I don't get how I'm supposed to fix this or why this happens. Adding `import de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin` doesn't seem to do anything?
> 
> ```
> * What went wrong:
> A problem occurred evaluating root project 'plugin-template'.
> > Failed to apply plugin class 'de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin'.
>    > Could not create plugin of type 'ForbiddenApisPlugin'.
>       > Could not initialize class de.thetaphi.forbiddenapis.gradle.ForbiddenApisPlugin
> ```

Interesting..do you who is pulling ForbiddenApisPlugin?
From `build.gradle` I dont see anything.

**Comment by saratvemulapalli** [comment]
Looks like OpenSearch is doing it via gradle.
https://github.com/opensearch-project/OpenSearch/search?q=ForbiddenApisPlugin

**Comment by dblock** [comment]
The current error is 

```
> Task :yamlRestTest

org.opensearch.path.to.plugin.RenameClientYamlTestSuiteIT > initializationError FAILED
    java.lang.NoClassDefFoundError: org/apache/logging/log4j/core/Layout
```

I don't get why this is happening.

@reta maybe you will have some ideas?

**Comment by dblock** [comment]
@AmiStrn any ideas how to fix this?

**Comment by reta** [comment]
@dblock oh I know what this is about

**Comment by AmiStrn** [comment]
Its gotta be something in the build.gradle file. It is pretty late on this side of the globe. I can take a look later on friday unless @reta has it pin pointed already:)

**Comment by reta** [comment]
@dblock I created this pull request to fix the issue https://github.com/dblock/opensearch-plugin-template-java/pull/1

**Comment by reta** [comment]
> Its gotta be something in the build.gradle file. It is pretty late on this side of the globe. I can take a look later on friday unless @reta has it pin pointed already:)

I bit more tricker, it is optional dependencies, which seems to be treated differently when `api "org.opensearch:opensearch:${opensearch_version}"` vs `  api project(':server')` :(

**Comment by dblock** [comment]
🤯  @reta I don't know why or how you knew how to fix this one but hat tip

**Comment by reta** [comment]
> exploding_head @reta I don't know why or how you knew how to fix this one but hat tip

It happens I run into that a few days ago while working on POC to move out the repository plugins :smile:

**Review by reta — APPROVED** [review]

**Review by AmiStrn — APPROVED** [review]
Thank you @reta for your valuable insight on this issue!

**Comment by dblock** [comment]
@CEHENKLE Can you please help with permissions here? We have a 2 people approvals required by the org, and there's not 2 people maintaining this repo.

**Comment by dblock** [comment]
@AmiStrn cool if I add @saratvemulapalli to PR reviewers to this project?
 
Maybe @reta is also interested?

File: build.gradle [hunk 1]
```diff
@@ -89,3 +93,7 @@ testClusters.integTest {
 run {
     useCluster testClusters.integTest
 }
+
+tasks.withType(RestIntegTestTask)*.configure {
+  classpath += files(project.configurations.runtimeClasspath.findAll { it.name.contains("log4j-core") })
+}
```
Comment (saratvemulapalli) [inline build.gradle:99]: Why is `log4j-core` special?

**Comment by reta** [comment]
> @AmiStrn cool if I add @saratvemulapalli to PR reviewers to this project?
> 
> Maybe @reta is also interested?

@dblock I am certainly in, thank you

File: build.gradle (hunk 1 above)
Comment (reta) [reply build.gradle:99]: @saratvemulapalli see please https://github.com/opensearch-project/opensearch-plugin-template-java/pull/16#issuecomment-1000536819, it is need but is optional

**Comment by dblock** [comment]
> > @AmiStrn cool if I add @saratvemulapalli to PR reviewers to this project?
> > Maybe @reta is also interested?
> 
> @dblock I am certainly in, thank you

Thanks. @AmiStrn LMK if it's ok to add these two

File: build.gradle (hunk 1 above)
Comment (saratvemulapalli) [reply build.gradle:99]: Thanks @reta , I did read through the comments on https://github.com/dblock/opensearch-plugin-template-java/pull/1. 
I still dont understand why :)

Comment (reta) [reply build.gradle:99]: There is something coming from the plugin apparently, I will create an issue shortly to figure out exactly what is going on, will ping you there, thanks!

**Comment by AmiStrn** [comment]
> Thanks. @AmiStrn LMK if it's ok to add these two with write permissions?

@saratvemulapalli and @reta are more than welcome:)

**Review by saratvemulapalli — APPROVED** [review]

**Comment by AmiStrn** [comment]
@dblock i think i can add them myself. is this ok in terms of how we should be doing things?

**Comment by AmiStrn** [comment]
@dblock I added @saratvemulapalli because he is in the org it seemed to work fine. However, im gonna need help with @reta  🙏

**Comment by dblock** [comment]
> @dblock I added @saratvemulapalli because he is in the org it seemed to work fine. However, im gonna need help with @reta 🙏

Exactly. I think org owners need to do something to bring in an external user for the first time. I’ll go find out how.

**Comment by dblock** [comment]
@reta you should have received an invite to this repo, please let me know if you haven't

**Comment by reta** [comment]
@dblock all good, got it and accepted it, thank you! ❤️

**Comment by dblock** [comment]
> @dblock all good, got it and accepted it, thank you! ❤️

Thank you for contributing!

**Pull request merged**